### ISSUE TITLE
Rule blocked: CORERULES-1636

### ISSUE THREAD

**Issue by FITZPST** [cdisc-org/cdisc-rules-engine#290]
- Attach test dataset files that can be used to test the new feature
- Replace the "###" in the issue title with the primary JIRA ticket number for the rule. 
- Fill in the following information

**Links to related JIRA Tickets**
/https://jira.cdisc.org/browse/CORERULES-1636

**Rule Information**
- **Standard**: ADaM
- **Rule ID**: Check number 5
- **Rule Description**:  For character flags with variable names that end in FL  (a suffix of FL) and that are not population flags has a value that is not Y, N or null

**Describe the problem**
Cannot parse metadata of all variables in a dataset for specific substring matches

**Describe the solution**
Ability to scan all variable names (i.e search all metadata variable_names  for specific substrings:  suffixes, prefixes and embedded strings. Once identified allow record level values of variables to be checked?

**Proposed rule logic**
```yaml
-name: variable_name *operating at metadata eval to select variables that meet criteria
 operator:  starts_with|end_with|contains  
 value : "FL" 
-operator:  not_contain  
 value :  ('Y' , 'N')
-name: variable_name 
 operator:  not_empty

Note operators starts_with/end_with already exist for record level operations,  need to apply at metdata level.
 
```

**Screenshots**
If applicable, add screenshots to help explain your problem and solution.
